index.js

```diff
@@ -223,7 +244,8 @@ DB.prototype.batch = function (rows, opts, cb) {
         batch.push(row)
         if (--pending === 0) done()
       } else if (row.type === 'del') {
-        self._del(key, xtend(opts, row), function (err, xrows) {
+        var xrow = xtend(opts, row)
+        self._getDocumentDeletionBatchOps(key, xrow, function (err, xrows) {
```

Comment:
We need to be careful here:

`del() --> _getDocumentDeletionBatchOps() --> batch() --> _getDocumentDeletionBatchOps()` - is that going to cause any weirdness / errors? Can we avoid this loop?